addons/project/controllers/portal.py

```diff
@@ -410,7 +410,7 @@ def _prepare_tasks_values(self, page, date_begin, date_end, sortby, search, sear
 
         Task = request.env['project.task']
         milestone_domain = AND([domain, [('allow_milestones', '=', True)], [('milestone_id', '!=', False)]])
-        milestones_allowed = Task.search_count(milestone_domain, limit=1) == 1
+        milestones_allowed = Task.sudo().search_count(milestone_domain, limit=1) == 1
```

Comment:
Adding a sudo here is not a good idea. The goal of the search_count is to check if out of the tasks that the user has access to, some have the 'milestone_allowed' field set to true. Using a sudo will check for all the task in the db, so its a change in the behavior. Sadly, I dont really have a better idea on how to solve the access error while keeping the current behavior. I suppose we could add a condition on the domain with the collaborators of the tasks, but iirc it is quite costly to search on.
@xavierbol Do you have a better suggestion ? :angel:

Reply:
@damrOdoo we could maybe move the code in L413-417 below the code at the line 423. 🤔 

What do you think? 🙂

Reply:
yup, good idea, lets go with that.

Reply:
@damrOdoo 
done with changes